File: J_se_les12/HW/Core/Logic/Add/AddBase.java
package Java.J_se_les12.HW.Core.Logic.Add;

import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleInput;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public abstract class AddBase {
    Phonebook phonebook;
    IConsoleInput input;

    public AddBase(Phonebook phonebook, IConsoleInput input) {
        this.phonebook = phonebook;
        this.input = input;
    }

    public abstract void add();
}

File: J_se_les13/HW/Core/Logic/Add/AddBase.cs
using System;
using HW.Core.Contact;
using HW.Core.Logic.Interfaces;
using HW.Core.Phonebook;
namespace HW.Core.Logic.Add
{
    public abstract class AddBase
    {
        public PhonebookV1 Phonebook { get; set; }
        public IConsoleInput Input { get; set; }
        protected AddBase(PhonebookV1 phonebook, IConsoleInput input)
        {
            Phonebook = phonebook;
            Input = input;
        }

        public abstract void Add();

    }
}